### PR TITLE
errorshow: Show hint when + is attempted on AbstractStrings

### PR DESCRIPTION
**Pull request by merryhime**
Addresses a part of #7512.

### CODE REVIEW

**Comment by KristofferC** [comment]
Nice idea! Could use a test.

File: base/errorshow.jl [hunk 1]
```diff
@@ -899,19 +899,26 @@ function show(io::IO, ip::InterpreterIP)
...
+    # (probably missing the operator)
+    # eg: (1 + 2)(3 + 4)
     if ex.f isa Number
         print(io, "\nMaybe you forgot to use an operator such as ")
         printstyled(io, "*, ^, %, / etc. ", color=:cyan)
         print(io, "?")
     end
+    # Display a hint in case the user tries to use the + operator on strings
+    # (probably attempting concatenation)
+    if (ex.f == +) && all(i -> i <: AbstractString, arg_types)
```
Comment (fredrikekre) [inline base/errorshow.jl:914]: Maybe put this in a separate function with it's own call to `Experimental.register_error_hint`?

Comment (merryhime) [reply base/errorshow.jl:923]: Done!

**Comment by merryhime** [comment]
Good idea - wrote a simple test.

**Comment by KristofferC** [comment]
Open an issue maybe?

**Pull request merged**